src/hash.ts

```diff
@@ -1,63 +1,44 @@
+import { ScriptInfo } from "./lua-scripts/hash";
 import { Context, RegionContext } from "./types"
 
-type ScriptKind = "limitHash" | "getRemainingHash" | "resetHash"
-
 /**
- * Loads the scripts to redises with SCRIPT LOAD if the first region context
- * doesn't have the kind of script hash in it. 
+ * Runs the specified script with EVALSHA using the scriptHash parameter.
  * 
- * @param ctx Regional or multi region context
- * @param script script to load
- * @param kind script kind
- */
-const setHash = async (
-  ctx: Context,
-  script: string,
-  kind: ScriptKind
-) => {
-  const regionContexts = "redis" in ctx ? [ctx] : ctx.regionContexts
-  const hashSample = regionContexts[0].scriptHashes[kind]
-  if (!hashSample) {
-    await Promise.all(regionContexts.map(async (context) => {
-      context.scriptHashes[kind] = await context.redis.scriptLoad(script)
-    }));
-  };
-}
-
-/**
- * Runds the specified script with EVALSHA if ctx.cacheScripts or EVAL
- * otherwise.
- * 
- * If the script is not found when EVALSHA is used, it submits the script
- * with LOAD SCRIPT, then calls EVALSHA again.
+ * If the EVALSHA fails, loads the script to redis and runs again with the
+ * hash returned from Redis.
  * 
  * @param ctx Regional or multi region context
- * @param script script to run
- * @param kind script kind
- * @param keys 
- * @param args 
+ * @param script ScriptInfo of script to run. Contains the script and its hash
+ * @param keys eval keys
+ * @param args eval args
  */
 export const safeEval = async (
   ctx: RegionContext,
-  script: string,
-  kind: ScriptKind,
+  script: ScriptInfo,
   keys: any[],
   args: any[],
 ) => {
-  if (!ctx.cacheScripts) {
-    return await ctx.redis.eval(script, keys, args);
-  };
-
-  await setHash(ctx, script, kind);
   try {
-    return await ctx.redis.evalsha(ctx.scriptHashes[kind]!, keys, args)
+    return await ctx.redis.evalsha(script.hash, keys, args)
   } catch (error) {
     if (`${error}`.includes("NOSCRIPT")) {
-      console.log("Script with the expected hash was not found in redis db. It is probably flushed. Will load another scipt before continuing.");
-      ctx.scriptHashes[kind] = undefined;
-      await setHash(ctx, script, kind)
-      console.log("  New script successfully loaded.")
-      return await ctx.redis.evalsha(ctx.scriptHashes[kind]!, keys, args)
+      console.log(
```

Comment:
Do we really when these to be printed by default? Maybe we should have a debug config

Reply:
I removed the logs, but kept the warning below. Warning should never be logged, so if it does I think we can print by default https://github.com/upstash/ratelimit-js/pull/120#discussion_r1751448003